src/openforms/formio/tests/validation/test_email.py

```diff
@@ -94,3 +98,64 @@ def test_email_with_plugin_validator(self):
                 )
 
                 self.assertFalse(is_valid)
+
+
+class EmailValidationDBTests(TestCase):
+    def test_require_email_verification(self):
+        submission = SubmissionFactory.create()
+        # create some *unrelated* verification records
+        ev = EmailVerificationFactory.create(
+            component_key="foo", email="foobar@example.com", verified=True
+        )
+        assert ev.submission != submission
+        EmailVerificationFactory(
+            submission=submission,
+            component_key="foo",
+            email="foobar@example.com",
+            verified=False,
+        )
+        EmailVerificationFactory(
+            submission=submission,
+            component_key="bar",
+            email="foobar@example.com",
+            verified=True,
+        )
+        EmailVerificationFactory(
+            submission=submission,
+            component_key="foo",
+            email="other@example.com",
+            verified=True,
+        )
+
+        component: Component = {
+            "type": "email",
+            "key": "foo",
+            "label": "Test",
+            "openForms": {
+                "requireVerification": True,
+            },
+        }
+        data: JSONValue = {"foo": "foobar@example.com"}
+
+        with self.subTest("not verified"):
+            is_valid, errors = validate_formio_data(
+                component, data, submission=submission
+            )
+
+            self.assertFalse(is_valid)
+            error = extract_error(errors, "foo")
+            self.assertEqual(error.code, "unverified")
+
+        with self.subTest("verified"):
+            # multiple successful verifications should not be a problem
```

Comment:
Tested this with multiple verification codes back and forth and with logic rules and indeed it's working as expected (since we allow this kind of behaviour).